src/main/java/picard/fingerprint/CrosscheckFingerprints.java

```diff
@@ -713,6 +757,33 @@ private Map<String, String> getStringStringMap(final File sampleMapFile, final S
         return sampleMap;
     }
 
+    private Map<Path, Path> getPathPathMap(final File indexMapFile, final String inputFieldName) {
+        final Map<String, String> indexStringMap = getStringStringMap(indexMapFile, inputFieldName);
+        final HashMap<Path, Path> indexPathMap = new HashMap<Path, Path>();
+        for (Map.Entry<String, String> entry: indexStringMap.entrySet()) {
+            Path inputPath = null;
```

Comment:
This was a bit tricky because of the limited scope of the try block. It's possible to refactor and put the `try` block around the loop, but then you miss the rich exception messages about which file caused the exception. Unless you have a good idea of how to preserve that while getting `final` variables?

Reply:
```suggestion
            final Path inputPath;
```

The point is that you can declare the variable (and make it final) without immediately assigning.  As long as you only assign it once, it can still be final.

Reply:
Oh neat, I didn't realize this was possible! Changed